packages/asset-service/src/generateAssetData/coingecko.ts

```diff
@@ -0,0 +1,81 @@
+import { adapters, avalancheChainId, ChainId, ethChainId, toAssetId } from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+import lodash from 'lodash'
+
+import { avax, ethereum } from './baseAssets'
+
+type Token = {
+  chainId: number
+  address: string
+  name: string
+  symbol: string
+  decimals: number
+  logoURI: string
+}
+
+type TokenList = {
+  name: string
+  logoURI: string
+  keywords: string[]
+  timestamp: string
+  tokens: Array<Token>
+}
+export async function getAssets(
+  chainId: ChainId,
+  overrideAssets: Array<Asset> = []
+): Promise<Asset[]> {
+  const { category, explorer, explorerAddressLink, explorerTxLink } = (() => {
+    switch (chainId) {
+      case ethChainId:
+        return {
+          category: adapters.chainIdToCoingeckoAssetPlatform(chainId),
+          explorer: ethereum.explorer,
+          explorerAddressLink: ethereum.explorerAddressLink,
+          explorerTxLink: ethereum.explorerTxLink
+        }
+      case avalancheChainId:
+        return {
+          category: adapters.chainIdToCoingeckoAssetPlatform(chainId),
+          explorer: avax.explorer,
+          explorerAddressLink: avax.explorerAddressLink,
+          explorerTxLink: avax.explorerTxLink
+        }
+      default:
+        throw new Error(`no coingecko token support for chainId: ${chainId}`)
+    }
+  })()
+
+  const { data } = await axios.get<TokenList>(`https://tokens.coingecko.com/${category}/all.json`)
+
+  return data.tokens.reduce<Array<Asset>>((prev, token) => {
+    try {
+      const assetId = toAssetId({ chainId, assetNamespace: 'erc20', assetReference: token.address })
```

Comment:
the string value is a const, the `as const` is intended to be used when referencing the underlying value of a string variable

```
const assetNamespace = 'erc20'
toAssetId({ chainId, assetNamespace: assetNamespace as const})
```

needed since the type of `assetNamespace` var is string, but we want the type to be inferred as the underlying constant value of `erc20`